### PR TITLE
Feature/M2 1393 connect activity flows to the API

### PR DESCRIPTION
**Pull request by IhorNesterenko99**

### CODE REVIEW

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 1]
```diff
@@ -75,8 +76,25 @@ export const Activities = () => {
         {activities?.length ? (
           activities.map((item: ActivityFormValues, index: number) => {
             const handleEdit = () => navigateToActivity(item.key ?? item.id);
-            //TODO: check if dependent activity flow is required to remove
-            const handleRemove = () => removeActivity(index);
+
+            const handleRemove = () => {
```
Comment (kdanilenka32) [inline src/modules/Builder/features/Activities/Activities.tsx:80]: can y move this outside map? y have (n^3) complexity

File: src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts [hunk 2]
```diff
@@ -62,7 +61,11 @@ export const useAppletData = () => {
       description: appletDescription,
       about: appletAbout,
       themeId: null, // TODO: create real themeId
-      activityFlows: [],
+      activityFlows: appletInfo?.activityFlows.map(({ key, ...flow }) => ({
+        ...flow,
```
Comment (kdanilenka32) [inline src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts:65]: can y remove 1 ...?

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 3]
```diff
@@ -75,8 +76,25 @@ export const Activities = () => {
...
+              const activityKey = item.id || item.key;
```
Comment (alex-i-e) [inline src/modules/Builder/features/Activities/Activities.tsx:81]: could you rename `item` to `activity` to be consistent and to have differensies between real items

File: src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx [hunk 4]
```diff
@@ -141,7 +141,11 @@ export const ActivityFlow = () => {
             {(listProvided) => (
               <Box {...listProvided.droppableProps} ref={listProvided.innerRef}>
                 {activityFlows.map((item, index) => (
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx:143]: please rename `item` to `activityFlow` or `flow`

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx [hunk 5]
```diff
@@ -36,7 +36,9 @@ export const ActivityFlowBuilder = () => {
   const { control, watch } = useFormContext();
   const { appletId, activityFlowId } = useParams();
   const activityFlows: AppletFormValues['activityFlows'] = watch('activityFlows');
-  const activityFlowIndex = activityFlows.findIndex((flow) => flow.id === activityFlowId);
+  const activityFlowIndex = activityFlows.findIndex(
+    (flow) => (flow.id || flow.key) === activityFlowId,
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:40]: could you move this function and the same function in ActivityFlowAbout.tsx, maybe make it more generic:
```
const checkOnIdOrKey = (checkId: string) => (entity: TypeThatHasIdAndKey) => (entity.id || entity.key) === checkId
...
const activityFlowIndex = activityFlows.findIndex(checkIdOrKey(activityOnFlowId));
```

**Review by muhurov — APPROVED** [review]

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx [hunk 6]
```diff
@@ -85,12 +86,15 @@ export const ActivityFlowBuilder = () => {
     move(source.index, destination.index);
   };
 
-  const activitiesIds = activities.reduce((acc: Record<string, ActivityFormValues>, activity) => {
-    const id = activity.id || activity.key || '';
-    acc[id] = activity;
+  const activitiesIdsObjects = activities.reduce(
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:89]: could you move this functuin into utils?

File: src/shared/state/Applet/Applet.schema.ts [hunk 7]
```diff
@@ -17,14 +17,24 @@ export type CreateAppletStateData = {
   key: keyof AppletSchema;
 };
 
+type ActivityFlowItem = {
+  activityId: string;
+  id: string;
+  order: number;
+  activityKey?: string;
```
Comment (alex-i-e) [inline src/shared/state/Applet/Applet.schema.ts:24]: do we really need to use activityId and activityKey? maybe assign `activityKey` to activityId as a fallback as you did before in other places?

File: src/modules/Builder/features/Activities/Activities.tsx (hunk 1 above)
Comment (IhorNesterenko99) [reply src/modules/Builder/features/Activities/Activities.tsx:80]: done

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx (hunk 5 above)
Comment (IhorNesterenko99) [reply src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:40]: done

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx (hunk 6 above)
Comment (IhorNesterenko99) [reply src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:89]: done

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**